seahub/work_weixin/views.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2012-2016 Seafile Ltd.
+# encoding: utf-8
+
+import uuid
+import json
+import logging
+import requests
+from oauthlib.common import add_params_to_uri
+from django.http import HttpResponseRedirect
+from django.utils.translation import ugettext as _
+
+from seahub.api2.utils import get_api_token
+from seahub import auth
+from seahub.utils import render_error
+from seahub.base.accounts import User
+# from social_django.models import UserSocialAuth
+from seahub.work_weixin.settings import WORK_WEIXIN_AUTHORIZATION_URL, WORK_WEIXIN_CORP_ID, \
+    WORK_WEIXIN_AGENT_ID, WORK_WEIXIN_REDIRECT_URI, WORK_WEIXIN_PROVIDER, \
+    WORK_WEIXIN_GET_USER_INFO_URL, WORK_WEIXIN_GET_USER_PROFILE_URL, WORK_WEIXIN_UID_PREFIX
+from seahub.work_weixin.utils import work_weixin_oauth_check, get_work_weixin_access_token, \
+    handler_work_weixin_api_response, update_work_weixin_user_info
+from seahub.utils.auth import gen_user_virtual_id
+from seahub.utils import get_service_url
+from seahub.auth.models import SocialAuthUser
+
+logger = logging.getLogger(__name__)
+
+
+# # uid = corpid + '_' + userid
+
+
+def work_weixin_oauth_login(request):
```

Comment:
1,  not allowed creeate unknow user:    Error, new user registration is not allowed, please contact administrator.
1,  not auto active: Your account is created successfully, please wait for administrator to activate your account.
1,  not enable oauth login: Feature disabled.
1,  enable oauth login but wrong congure: Error, please contact administrator.